Std/Data/BitVec/Lemmas.lean

```diff
@@ -117,6 +117,10 @@ private theorem lt_two_pow_of_le {x m n : Nat} (lt : x < 2 ^ m) (le : m ≤ n) :
   else
     simp [h]
 
+@[simp]
+theorem BitVec.ofInt_ofNat : BitVec.ofInt w (no_index (OfNat.ofNat n : Int)) = BitVec.ofNat w n := by
```

Comment:
```suggestion
theorem ofInt_ofNat : BitVec.ofInt w (no_index (OfNat.ofNat n : Int)) = BitVec.ofNat w n := by
```
I am also unsure what the no_index is.

Reply:
It fixes a problem in the `simp` unification. If we directly write `BitVec.ofInt w (OfNat.ofNat n : Int)`, the data structure that `simp` uses to find patterns (a [discrimination tree](https://www.cs.cmu.edu/~fp/courses/99-atp/lectures/lecture28.html)) does not index the term for unclear reasons. @alexkeizer showed me this trick.

Reply:
Note that I'm not a 100% sure what `no_index` does either. I just copied this from a similar lemma in mathlib, after observing that the lemma without `no_index` would fail to trigger with `simp` on literals.

We should probably explain why we added `no_index` in the PR description

Reply:
OK. Someone should accept my change, btw.

Reply:
(I've accepted the suggestion, but unresolved to retain the `no_index` discussion)